kiwi/iso_tools/xorriso.py

```diff
@@ -85,23 +89,40 @@ def init_iso_creation_parameters(self, custom_args=None):
         self.iso_parameters += [
             '-joliet', 'on', '-padding', '0'
         ]
-        loader_file = self.boot_path + '/loader/isolinux.bin'
 
-        syslinux_lookup_paths = [
-            '/usr/share/syslinux', '/usr/lib/syslinux/modules/bios',
-            '/usr/lib/ISOLINUX'
-        ]
-        mbr_file = Path.which('isohdpfx.bin', syslinux_lookup_paths)
-        if not mbr_file:
-            raise KiwiIsoToolError(
-                'isohdpfx.bin not found in {0}'.format(syslinux_lookup_paths)
-            )
+        if Defaults.is_x86_arch(self.arch):
+            if efi_mode:
+                loader_file = os.sep.join(
+                    [
+                        self.boot_path, 'loader',
+                        Defaults.get_isolinux_bios_grub_loader()
+                    ]
+                )
+                mbr_file = os.sep.join(
+                    [self.source_dir, self.boot_path, '/loader/boot_hybrid.img']
+                )
+                self.iso_loaders += [
+                    '-boot_image', 'grub', 'bin_path={0}'.format(loader_file),
+                    '-boot_image', 'grub', 'grub2_mbr={0}'.format(mbr_file),
+                    '-boot_image', 'grub', 'grub2_boot_info=on'
+                ]
+            else:
+                loader_file = self.boot_path + '/loader/isolinux.bin'
+                mbr_file = Path.which(
+                    'isohdpfx.bin', Defaults.get_syslinux_search_paths()
+                )
```

Comment:
This `Path.which` call returns `None` if not found, I guess this is always an issue. Probably we could add something like `raise_not_found` in `Path.which` add raise an exception of path not found if set to true. What you think? I discovered that playing around with Arch an it took me while to figure out which path was missing by looking at the xorriso command line. Probably providing an exception stating the missing path is more clear.

I would be fine if we handle that in a separate PR and probably find out all the places where it could make sense. Probably there are other `Path.which` in the code where this applies.

Reply:
When I implemented that it was done on purpose. So in the code you will always find the result of Path.which to be followed by a conditional check. I'm ok if we refactor that to be exception based like Command but for the moment I don't see a high demand and yes if this will be done then in a separate PR which will probably quite big because it's used pretty often in the code

Reply:
Fine with me, we can leave it as is for the time being and reconsider it if at some point the topic raises again. I am also fine leaving `mbr_file` without a conditional check since it is just used straight after the `Path.which` call.